src/ensembl/src/content/app/browser/browserActions.ts

```diff
@@ -79,10 +82,14 @@ export const setDataFromUrlAndSave: ActionCreator<
   browserStorageService.saveActiveGenomeId(payload.activeGenomeId);
   chrLocation &&
     browserStorageService.updateChrLocation({ [activeGenomeId]: chrLocation });
-  activeEnsObjectId &&
+
+  if (activeEnsObjectId) {
     browserStorageService.updateActiveEnsObjectIds({
       [activeGenomeId]: activeEnsObjectId
     });
+
+    dispatch(updatePreviouslyViewedObjectsAndSave());
+  }
```

Comment:
Having `updatePreviouslyViewedObjectsAndSave` both in `setDataFromUrlAndSave` and in `changeObject` makes it being called twice. It should either be called in `changeBrowserLocation` and `changeObject` or only in `setDataFromUrlAndSave`. Not sure which is better.

Reply:
Removed it from `setDataFromUrlAndSave ` and added it to `changeBrowserLocation`. Keeping it only on `setDataFromUrlAndSave` doesn't seem to work when I change the focus from the zmenu.